### PR TITLE
DAO-844: add checksum validation

### PR DESCRIPTION
**Pull request by sleyter93**
# Description
Checksum validation for address in delegate and proposal views.
Screenshot 2024-11-27 at 12 58 05 AM
Screenshot 2024-11-27 at 12 58 46 AM

### CODE REVIEW

File: src/app/proposals/create/TreasuryWithdrawProposalForm.tsx [hunk 1]
```diff
@@ -35,37 +35,45 @@ import { useForm } from 'react-hook-form'
...
-    message: `The minimum amount for ERC20 tokens is ${rifMinimumAmount}.0`,
-    path: ['amount'],
-  })
-  .refine(data => data.tokenAddress !== zeroAddress || Number(data.amount) >= rbtcMinimumAmount, {
-    message: `The minimum amount for RBTC is ${rbtcMinimumAmount}`,
-    path: ['amount'],
-  })
-
 export const TreasuryWithdrawProposalForm = () => {
+  const { chainId } = useAccount()
```
Comment (Freshenext) [inline src/app/proposals/create/TreasuryWithdrawProposalForm.tsx:44]: I am seeing this and something pops to my head: why do we need to move this logic within React ecosystem and not keep it outside?

**Review by Freshenext — DISMISSED** [review]

File: src/app/proposals/create/TreasuryWithdrawProposalForm.tsx (hunk 1 above)
Comment (sleyter93) [reply src/app/proposals/create/TreasuryWithdrawProposalForm.tsx]: It was required just for chainId. I've updated to use CHAIN_ID constant, so I return the FormSchema outside the react.

**Review by rodrigoncalves — CHANGES_REQUESTED** [review]
Looks good just a suggestion of minor change.

**Review by rodrigoncalves — APPROVED** [review]
PR works fine. Just did a minor change clearing the error when changing the address.

**Pull request merged**